cabal.project

```diff
@@ -192,8 +195,8 @@ source-repository-package
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-prelude
-  tag: bb4ed71ba8e587f672d06edf9d2e376f4b055555
-  --sha256: 00h10l5mmiza9819p9v5q5749nb9pzgi20vpzpy1d34zmh6gf1cj
+  tag: 533aec85c1ca05c7d171da44b89341fb736ecfe5
+  --sha256: 0z2y3wzppc12bpn9bl48776ms3nszw8j58xfsdxf97nzjgrmd62g
```

Comment:
I'm concerned about this and the other commit updates. Are these pointed to by the node at consensus/feature/utxo-hd? if not, it might not be right to do this update as this might include not expected changes.

Reply:
I see two options: 1. bumping dependencies, and 2. generating the reports from this branch without ever merging it into feature/utxo-hd. Spoiler: I am in favor of bumping deps, because the risks seem manageable (but as of yet, I have little intuition for the non-static breakages that cardano-ledger and cardano-prelude updates could cause).

**1. Bumping dependencies**

At a glance it seems there are two potential commits in ledger that might break things invisibly:
1. f0d39d9 New CBOR instance for PState
2. ee64d21 Bumped plutus

I think that it is sufficient to update cardano-ledger up to the point of the nix-packages update commit (6a15bed), which is just before `ee64d21 Bumped plutus`. So it is seems feasible and less risky to set the cardano-ledger dependency to 6a15bed, and keep plutus as is.

If we then choose to keep cardano-prelude in sync with ledger we'd still need to bump it to 533aec85. The version that is being used currently here, and the feature/utxo-hd branch at cardano-node is quite old. March 30, 2021, and September 29, 2021. Naively it seems that any breaking changes from prelude should break the build here, so I'd say there is less risk bumping cardano-prelude.

**1. Not merging**

This is fine for generating the coverage reports for now, however, it adds annoying overhead when:
1. We want to regenerate the reports (for example after adding tests).
2. We want to integrate coverage reporting into CI.

Reply:
Reverting dependencies to those on feature/utxo-hd seems to work.